mev-boost-rs/src/relay_mux.rs

```diff
@@ -146,12 +147,13 @@ impl BlindedBlockProvider for RelayMux {
             return Err(Error::NoBids)
         }
 
-        // for now, break any ties by picking the first bid,
-        // which currently corresponds to the fastest relay
-        let (best_index, rest) = best_indices.split_first().unwrap();
+        // if multiple indices with same bid value, break tie by randomly picking one
+        let mut rng = rand::thread_rng();
+        let best_index = &best_indices[rng.gen_range(0..best_indices.len())];
+        let rest = best_indices.iter().filter(|i| **i != *best_index);
```

Comment:
this will work, although what do you think about just shuffling the indices first and then splitting away the first index? 

e.g. https://rust-random.github.io/rand/rand/seq/trait.SliceRandom.html#tymethod.shuffle

```rust
let mut rng = rand::thread_rng();
best_indices.shuffle(&mut rng);
// keep the rest of the existing code with `split_first`
```

I prefer this because we don't need to worry ourselves about things like the length of the indices (`best_indices.len()`) or bothering with indexing semantics (`0..best_indices.len()`) and can avoid that call to `filter` to split off the relevant indices

Reply:
This seems better way!